google-cloud-apis/java-apis/maven-apis/src/com/google/cloud/tools/intellij/cloudapis/maven/MavenCloudApiActionDecorator.java

```diff
@@ -0,0 +1,110 @@
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleManager;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.JavaVersion;
+
+/**
+ * {@link CloudApiActionDecoratorExtension} extension point that checks for Java projects version if
+ * App Engine is used, and ensures Maven dependencies support is only available for Maven projects.
+ * Otherwise, action is disabled and help text is shown.
+ */
+public class MavenCloudApiActionDecorator implements CloudApiActionDecoratorExtension {
+
+  @Override
+  public void decorate(AnActionEvent e) {
+    boolean addLibrariesEnabled = false;
+    if (e.getProject() != null) {
+      Set<CloudLibrariesModuleSupportType> moduleSupportTypes =
+          Stream.of(ModuleManager.getInstance(e.getProject()).getModules())
+              .map(this::checkModuleForAddCloudLibraries)
+              .collect(Collectors.toSet());
+
+      addLibrariesEnabled = moduleSupportTypes.contains(CloudLibrariesModuleSupportType.SUPPORTED);
+      if (!addLibrariesEnabled) {
+        // update message to hint what is missing.
+        for (CloudLibrariesModuleSupportType supportType : moduleSupportTypes) {
+          switch (supportType) {
+            case MAVEN_REQUIRED:
+              e.getPresentation()
+                  .setDescription(
+                      MavenCloudApisMessageBundle.message(
+                          "cloud.libraries.menu.action.maven.required.description"));
+              e.getPresentation()
+                  .setText(
+                      MavenCloudApisMessageBundle.message(
+                          "cloud.libraries.menu.action.disabled.maven.text"));
+              break;
+            case APPENGINE_JAVA8_REQUIRED:
+              e.getPresentation()
+                  .setDescription(
+                      MavenCloudApisMessageBundle.message(
+                          "cloud.libraries.menu.action.gae.java8.required.description"));
+              e.getPresentation()
+                  .setText(
+                      MavenCloudApisMessageBundle.message(
+                          "cloud.libraries.menu.action.disabled.java8.text"));
+              break;
+          }
+        }
+      } else {
+        // standard message for a supported module action.
+        e.getPresentation()
+            .setDescription(
+                GoogleCloudApisMessageBundle.message("cloud.libraries.menu.action.description"));
+        e.getPresentation()
+            .setText(
+                GoogleCloudApisMessageBundle.message("cloud.libraries.menu.action.active.text"));
+      }
+    }
+
+    e.getPresentation().setEnabled(addLibrariesEnabled);
+  }
+
+  private enum CloudLibrariesModuleSupportType {
+    SUPPORTED,
+    MAVEN_REQUIRED,
+    APPENGINE_JAVA8_REQUIRED
+  }
+
+  private CloudLibrariesModuleSupportType checkModuleForAddCloudLibraries(Module module) {
```

Comment:
nit: `checkAddCloudLibrariesSupport` (module is implied by the param, and adds the word `support` at the end to indicate what its checking)